extensions/markdown-language-features/src/features/previewManager.ts

```diff
@@ -197,6 +208,26 @@ export class MarkdownPreviewManager extends Disposable implements vscode.Webview
 		return this.registerDynamicPreview(preview);
 	}
 
+	public toggleStaticPreview(uri: vscode.Uri | undefined) {
+		const preview = this._activePreview;
+		if (preview instanceof StaticMarkdownPreview) {
+			vscode.commands.executeCommand('markdown.showTextEditor');
+		} else {
+			vscode.commands.executeCommand('markdown.showStaticPreview', uri);
+		}
+	}
+
+	public openStaticPreview(
+		uri: vscode.Uri,
+	): void {
+		vscode.commands.executeCommand('vscode.openWith', uri, this.customEditorViewType);
+	}
+
+	public openTextEditor(): void {
+		vscode.commands.executeCommand('workbench.action.closeActiveEditor');
```

Comment:
I _think_ this should be unnecessary - vscode.openWith should reopen the file using the right editor type

Reply:
It's really weird in that opening the 'default' viewtype seems to open in a new tab. You can also try commenting/uncommenting this line to double check this.

Reply:
openWith doesn't perform a reopen, but instead opens a new pinned editor. This behavior recently changed as prior to that they were preview editors, you can read more here https://github.com/microsoft/vscode/issues/123360